Azure.MgmtSdk.Analyzers/DataPropertyEtagAnalyzer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System;
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+
+namespace Azure.MgmtSdk.Analyzers
+{
+    /// <summary>
+    /// Analyzer to check resource type 'ResouceType'.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DataPropertyEtagAnalyzer : DiagnosticAnalyzer
+    {
+        protected static readonly string Title = "Improper public data property";
```

Comment:
Since we're not sure about the real data type, we'd better to use a softer tune here, like:
- Title: `Potential improper data type of propert`.
- MessageFormat: `Property `{0}` looks like an ETag`
- Description: `Check the real return value and consider changing the type to "ETag".`